iroh-metrics/src/core.rs

```diff
@@ -138,30 +172,28 @@ where
 /// Observing is for distribution metrics, when multiple observations are combined in a
 /// single metric value.
 #[allow(dead_code)]
-pub(crate) fn observe<M>(c: Collector, m: M, v: f64)
+pub async fn observe<M>(c: &str, m: M, v: f64)
 where
     M: HistogramType + std::fmt::Display,
 {
     if CORE.is_enabled() {
-        match c {
-            Collector::Iroh => CORE.iroh_metrics().observe(m, v),
-            Collector::Magicsock => CORE.magicsock_metrics().observe(m, v),
-            Collector::Netcheck => CORE.netcheck_metrics().observe(m, v),
-            Collector::Portmap => CORE.portmap_metrics().observe(m, v),
-        };
+        let cc = c.to_string();
+        let mn = m.name().to_string();
+        tokio::task::spawn(async move {
```

Comment:
why spawn if you are in an async fn?

Reply:
Yeah this passes only because it's not really used. Have to remove the async.